ledger-core/v2/insolar/gen/reference.go

```diff
@@ -21,24 +22,31 @@ func getUnique() uint32 {
 	return atomic.AddUint32(&uniqueSeq, 1)
 }
 
-// ID generates random id.
-func ID() reference.Local {
+func generateUniqueID(pn pulse.Number) reference.Local {
 	var id reference.Local
 
 	f := fuzz.New().NilChance(0).Funcs(func(id *reference.Local, c fuzz.Continue) {
 		var hash [reference.LocalBinaryHashSize]byte
 		c.Fuzz(&hash)
 		binary.BigEndian.PutUint32(hash[reference.LocalBinaryHashSize-4:], getUnique())
 
-		pn := PulseNumber()
-
 		*id = reference.NewRecordID(pn, reference.BytesToLocalHash(hash[:]))
 	})
 	f.Fuzz(&id)
 
 	return id
 }
 
+// ID generates random id.
+func ID() reference.Local {
+	return generateUniqueID(PulseNumber())
+}
+
+func UniqueIDWithPulse(pn insolar.PulseNumber) reference.Local {
```

Comment:
Is it really uniq? As I remember Fuzz is obligatory give you uniq numbers. It may be confusing

Reply:
```
binary.BigEndian.PutUint32(hash[reference.LocalBinaryHashSize-4:], getUnique())
```